2-0-data-structures-and-algorithms/src/test/java/com/bobocode/array_list/ArrayListTest.java

```diff
@@ -223,19 +275,31 @@ public void getElementByIndexEqualToListSize() {
     @Test
     @Order(21)
     public void setElementByIndex() {
-        arrayList = ArrayList.of(15, 69, 58, 78);
+        try {
+            getFieldByName("elementData").set(arrayList, new Object[]{15, 69, 58, 78});
+            getFieldByName("size").set(arrayList, 4);
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+        }
 
         arrayList.set(2, 10);
 
-        assertThat(arrayList.get(2)).isEqualTo(10);
-        assertThat(arrayList.get(3)).isEqualTo(78);
-        assertThat(arrayList.size()).isEqualTo(4);
+        Object[] elements = (Object[]) getObjectByName("elementData");
+        int size = (int) getObjectByName("size");
+
+        assertThat(elements[2]).isEqualTo(10);
+        assertThat(elements[3]).isEqualTo(78);
+        assertThat(size).isEqualTo(4);
     }
 
     @Test
     @Order(22)
     public void setElementByIndexEqualToSize() {
```

Comment:
@erneman1 let's rename this test method to `setElementByIndexThrowsExceptionWhenIndexIsOutOfBound`